pkg/kubelet/dockertools/docker_test.go

```diff
@@ -209,9 +209,100 @@ func TestMatchImageTagOrSHA(t *testing.T) {
 			Image:  "myimage@sha256:2208",
 			Output: false,
 		},
+		{
+			// mismatched ID is ignored
+			Inspected: dockertypes.ImageInspect{
+				ID: "sha256:2208f7a29005d226d1ee33a63e33af1f47af6156c740d7d23c7948e8d282d53d",
+			},
+			Image:  "myimage@sha256:0000f7a29005d226d1ee33a63e33af1f47af6156c740d7d23c7948e8d282d53d",
+			Output: false,
+		},
+		{
+			// invalid digest is ignored
+			Inspected: dockertypes.ImageInspect{
+				ID: "sha256:unparseable",
+			},
+			Image:  "myimage@sha256:unparseable",
+			Output: false,
+		},
+		{
+			// v1 schema images can be pulled in one format and returned in another
```

Comment:
That test did fail.

On Wed, Sep 14, 2016 at 1:32 PM, Andy Goldstein <notifications@github.com>
wrote:

> In pkg/kubelet/dockertools/docker_test.go
> <https://github.com/kubernetes/kubernetes/pull/32627#discussion_r78795950>
> :
>
> > +			Inspected: dockertypes.ImageInspect{
> > +				ID: "sha256:2208f7a29005d226d1ee33a63e33af1f47af6156c740d7d23c7948e8d282d53d",
> > +			},
> > +			Image:  "myimage@sha256:0000f7a29005d226d1ee33a63e33af1f47af6156c740d7d23c7948e8d282d53d",
> > +			Output: false,
> > +		},
> > +		{
> > +			// invalid digest is ignored
> > +			Inspected: dockertypes.ImageInspect{
> > +				ID: "sha256:unparseable",
> > +			},
> > +			Image:  "myimage@sha256:unparseable",
> > +			Output: false,
> > +		},
> > +		{
> > +			// v1 schema images can be pulled in one format and returned in another
>
> Make a copy of this test case, change the value in RepoDigests so it's not
> a match, and the test will pass with the "double digest" line, and fail
> once you change one digest to d.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/32627/files/95cee6e53b91be6c7b5e37939c6374a37aa7e505#r78795950>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABG_pxOA5fM_3Y3-oXQFj54-C86qFeMBks5qqC-ygaJpZM4J8SYk>
> .
>